<table align="left">
  <td>
    <a href="https://colab.research.google.com/github/ageron/handson-ml3/blob/main/tools_numpy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>
  </td>
  <td>
    <a target="_blank" href="https://kaggle.com/kernels/welcome?src=https://github.com/ageron/handson-ml3/blob/main/tools_numpy.ipynb"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" /></a>
  </td>
</table>

### [Video de apoyo a la lectura interactiva y experimental de este cuaderno]()

### [Vínculo al programa del curso:  ]()



Un reconocimiento a mis estudiantes que han construido conmigo este saber pedagógico:

<img src = ''> 



Reconocimiento a las profesoras destacadas en enseñanza de la matemática en Colombia


In [ ]:
import pandas as pd
import numpy as np

# Cargar la lista de estudiantes desde el archivo CSV
path = 'C:/Users/marco/Documentos/docencia/groups_list/fundamentos_aritmetica.csv'
estudiantes = pd.read_csv(path)
estudiantes['Parcial'] = estudiantes['Parcial'].astype(float)  # Asegurar que la columna 'Parcial' es float

# Convertir la columna 'Nombre' a una lista de Python
lista_estudiantes = estudiantes['Nombre'].tolist()

# Función para llamar a un estudiante aleatoriamente y registrar su nota de seguimiento
def asistencia():
    global lista_estudiantes
    if lista_estudiantes:
        estudiante_aleatorio = np.random.choice(lista_estudiantes)
        print(f"Llamando a {estudiante_aleatorio}...")
        asistencia = input(f"¿Asistió {estudiante_aleatorio}? (s/n): ").strip().lower()
        
        # Validación de entrada
        if asistencia == 's':
            # Obtener el índice del estudiante en el DataFrame
            indice = estudiantes[estudiantes['Nombre'] == estudiante_aleatorio].index[0]
            # Incrementar su valor en la columna 'Parcial'
            estudiantes.at[indice, 'Parcial'] += 0.3
            print(f"Actualizado: {estudiante_aleatorio} ahora tiene {estudiantes.at[indice, 'Parcial']} en la columna 'Parcial'.")
        
        elif asistencia != 'n':
            print("Entrada inválida. Por favor, introduce 's' para sí o 'n' para no.")
            return  # Volver a preguntar para el mismo estudiante si la entrada es inválida
        
        # Remover al estudiante de la lista para no llamarlo de nuevo
        lista_estudiantes.remove(estudiante_aleatorio)
    else:
        print("Se han llamado a todos los estudiantes del grupo completo.")

# Función para guardar cambios al final de la sesión
def guardar_cambios():
    estudiantes.to_csv(path, index=False)
    print("Cambios guardados en el archivo CSV.")

# Ejemplo de uso
while lista_estudiantes:
    asistencia()

# Guardar los cambios una vez que todos los estudiantes hayan sido llamados
guardar_cambios()



Llamando a ROSILLO TEJADA VANESA...
Cambios guardados en el archivo CSV.
Llamando a QUINONEZ CERVANTES MARIANA...



# Principios de mi labor docente

## Modelo general de clase: [aula invertida](https://www3.gobiernodecanarias.org/medusa/ecoescuela/pedagotic/aprendizaje-invertido-flipped-classroom/)

<img src = 'https://resources.maneuveringthemiddle.com/wp-content/uploads/2020/07/04212706/1-5.png' width = 400>  



# Metodología de enseñanza basada en problemas (ABP)



<img src = 'https://www.unir.net/wp-content/uploads/2020/07/aprendizaje-problemas2-1-1-1-1.jpeg' width = 300>

# Desarrollo del pensamiento computacional con python y cuadernos Jupyter  orientado a la formación en ciencia de datos 




<img src = 'https://i.ytimg.com/vi/mcG_wcR4e6M/maxresdefault.jpg' width = 400>

# Principio de fomento a la ciencia abierta, enseñar en el acceso a los datos públicos y el uso de software de código abierto  



<img src = 'https://unesdoc.unesco.org/in/rest/Thumb/image?id=p%3A%3Ausmarcdef_0000379949_spa&author=UNESCO&title=Recomendaci%C3%B3n+de+la+UNESCO+sobre+la+Ciencia+Abierta&year=2021&publisher=UNESCO&TypeOfDocument=UnescoPhysicalDocument&mat=PGD&ct=true&size=512&isPhysical=1' width = 200>

El enfoque STEAMS en educación preescolar, primaria y secundaria es una extensión del enfoque STEM (Ciencia, Tecnología, Ingeniería y Matemáticas) y su variante STEAM, que incorpora el Arte. 

El acrónimo STEAMS incluye:



1. **Science (Ciencia):** Fomenta la curiosidad por el mundo natural y el desarrollo de habilidades científicas, como la observación, experimentación y análisis.



2. **Technology (Tecnología):** Integra herramientas tecnológicas para potenciar el aprendizaje y la resolución de problemas, promoviendo la alfabetización digital desde edades tempranas.



3. **Engineering (Ingeniería):** Introduce el pensamiento lógico y la resolución de problemas estructurados, fomentando habilidades como el diseño y la construcción de soluciones.



4. **Arts (Arte):** Añade creatividad y pensamiento crítico, permitiendo a los estudiantes expresarse y explorar el aprendizaje a través de formas artísticas (plásticas, música, teatro).



5. **Mathematics (Matemáticas):** Refuerza las habilidades numéricas y el razonamiento lógico, fundamentales en la resolución de problemas y comprensión de conceptos abstractos.



6. **Social Sciences (Ciencias Sociales):** Incorpora la comprensión del entorno social, político, cultural y económico, promoviendo el desarrollo de la conciencia crítica sobre el mundo y su historia, además de habilidades de empatía y ciudadanía.



El enfoque STEAMS pretende desarrollar un aprendizaje **interdisciplinario** que integre estas áreas para fomentar un pensamiento más holístico y práctico. 

Se busca que los estudiantes no solo aprendan conocimientos técnicos, sino también cómo aplicarlos creativa y críticamente en situaciones reales, promoviendo habilidades como el trabajo en equipo, la comunicación y la innovación. 

En educación preescolar, primaria, secundaria y superior, se adapta a las diferentes etapas de desarrollo cognitivo y emocional de los estudiantes, promoviendo el aprendizaje activo y contextualizado.  



<img src = 'https://www.campuseducacion.com/blog/wp-content/uploads/2023/03/foto01-1.jpg' width = 400>

# Qué estamos haciendo por la internalización en esta clase y este curso  

Para presentar tus acciones como docente de matemáticas en la Universidad de Antioquia con miras a la **internacionalización** de tu práctica docente y el desarrollo de competencias internacionales en tus estudiantes, te sugiero estructurar tu enfoque en los siguientes aspectos clave:

### 1. **Visión Global en la Formación Académica**


   - **Competencias Internacionales**: Asegúrate de que el contenido de tus cursos fomente habilidades que sean reconocidas globalmente. Al educar a los estudiantes en **Python, LaTeX, Markdown y Jupyter Notebooks** y en la **introducción a la ciencia de datos y técnicas de Machine Learning**, estás proporcionando competencias altamente demandadas en los sectores tecnológicos y académicos a nivel internacional.


   - **Relevancia de los Contenidos**: Destaca cómo los temas que enseñas, como el **desarrollo del pensamiento computacional** y la **ciencia de datos**, están alineados con las necesidades globales actuales en diversas industrias, desde la tecnología hasta las ciencias naturales y sociales.


   - **Referencias a Estándares Internacionales**: Puedes vincular tu enseñanza a estándares internacionales de competencias en programación y ciencia de datos (por ejemplo, frameworks del **IEEE, ACM**, y estándares de la **data science** o la **computación en la nube**).

### 2. **Innovación Pedagógica con Enfoque Internacional**


   - **Aula Invertida y ABP**: Presenta cómo tu enfoque de **aula invertida** y **aprendizaje basado en problemas (ABP)** permite a los estudiantes trabajar de manera autónoma y colaborativa, lo que los prepara para escenarios laborales internacionales que requieren autoaprendizaje y habilidades de resolución de problemas en equipo.


   - **Uso de Jupyter Notebooks**: El uso de **Jupyter Notebooks** en tus clases no solo facilita la **integración de código, matemáticas y texto**, sino que también es una plataforma utilizada globalmente en ambientes académicos y profesionales, lo que hace a tus estudiantes más competitivos internacionalmente.

### 3. **Tecnología y Herramientas de Uso Global**
   - **Lenguaje Python**: Python es uno de los lenguajes de programación más usados a nivel mundial, tanto en la academia como en la industria. Al formar a tus estudiantes en Python para ciencia de datos y machine learning, los preparas para operar en entornos tecnológicos globales.
   - **Herramientas de Ciencia Abierta**: Resalta cómo herramientas como **GitHub**, **Google Colab** y otras plataformas de código abierto están alineadas con el movimiento de **ciencia abierta**, el cual es internacionalmente reconocido y permite que los estudiantes colaboren y compartan proyectos a nivel global.

### 4. **Investigación y Publicaciones Internacionales**


   - **Promoción de la Investigación**: Detalla cómo promueves que tus estudiantes participen en proyectos de **investigación aplicada**, con posibilidad de publicación en revistas internacionales. Esto contribuye a la reputación internacional de la Universidad de Antioquia.


   - **Colaboraciones Internacionales**: Si has colaborado con investigadores de otras instituciones o universidades en el extranjero, menciona estos proyectos y cómo han enriquecido tu enseñanza y proporcionado a los estudiantes una visión más globalizada.

### 5. **Uso del Inglés para la Internacionalización**
   - **Comunicación Científica en Inglés**: Fomentar el aprendizaje del inglés como lengua científica es clave para que tus estudiantes puedan acceder a publicaciones internacionales, conferencias, y material académico que no está disponible en español. Esto es esencial para que adquieran competencias que les permitan desempeñarse en cualquier contexto internacional.
   - **Material Bilingüe**: Al generar material educativo en Python, LaTeX y Markdown, considera ofrecer algunos de estos recursos en inglés, para que los estudiantes se familiaricen con el vocabulario técnico y científico internacional.

### 6. **Certificaciones y Estándares Internacionales**
   - **Participación en Certificaciones Internacionales**: Impulsa que los estudiantes busquen certificaciones internacionales en Python o ciencia de datos (como las ofrecidas por **Coursera, edX,** o **Google**). Estas certificaciones no solo validan sus competencias, sino que también incrementan el reconocimiento de la Universidad de Antioquia en el contexto internacional.
   - **Alineación con Acreditaciones Internacionales**: Vincula tu trabajo docente con los estándares de acreditación y certificación que garantizan que la universidad siga siendo reconocida internacionalmente. Puedes vincularte con organizaciones internacionales que promuevan la excelencia en la educación y tecnología, como **ABET** para la acreditación en ingeniería y ciencias computacionales.

### 7. **Proyectos Abiertos y Ciencia Reproducible**
   - **GitHub y Open Science**: La utilización de **GitHub** para almacenar los proyectos y materiales de enseñanza en repositorios públicos permite que otros investigadores y educadores internacionales puedan revisar y replicar tu trabajo. Esto contribuye a la creación de una red de colaboración científica abierta.
   - **Transparencia y Reproducibilidad**: La ciencia de datos y machine learning están profundamente ligados a la reproducibilidad. Asegura que tus estudiantes desarrollen prácticas de transparencia y documentación en la investigación, que son pilares en la comunidad científica global.



### 8. **Participación en Redes y Conferencias Internacionales**
   - **Presentación en Conferencias Internacionales**: Destaca cómo tu participación en conferencias internacionales (como la **ASCOLFA** o eventos sobre educación en matemáticas) contribuye al prestigio de la universidad y permite que tus prácticas sean evaluadas y reconocidas a nivel global.
   - **Promoción de Estudiantes en Eventos Internacionales**: Impulsa la participación de tus estudiantes en competiciones internacionales de programación o hackathons, como los organizados por **Google**, **ACM** o **Kaggle**. Esto no solo desarrolla sus habilidades, sino que también visibiliza el talento de la Universidad de Antioquia a nivel global.

---

Con esta estrategia, presento mis acciones docentes en términos de competencias internacionales, innovación pedagógica, tecnología, investigación y participación en redes globales, asegurando que tu práctica esté alineada con los estándares internacionales y que tus estudiantes estén preparados para competir en contextos internacionales.

## Fomento de la lectura literaria asociada a Ingeniería agropecuaria y licenciatura en Matemáticas

[Aprender a obtener el sí sin ceder]()

## Título de la clase: Conjetura: Sobre la distribución de los primos en los números naturales que terminan en cifra 1  

[]()

### Objetivo de aprendizaje 


1. Conocimiento Teórico de Matemáticas

Indicadores:

* Comprensión de conceptos fundamentales de: .
* Capacidad para resolver ecuaciones: .
* Habilidad para: .
* Capacidad de leer, escribir e interpretar utilizando la notación y terminología matemática propia de esta clase.
* Capacidad para aplicar resultados científicos en problemas.



2. Habilidad en Programación con Python

Indicadores:

* Capacidad para escribir y ejecutar scripts básicos en Python.
* Comprensión y uso de estructuras de control (condicionales, bucles).
* Uso de funciones y módulos en Python.
* Manejo de estructuras de datos (listas, diccionarios, tuplas, nd-array, DataFrame, set).
* Habilidad para depurar y solucionar errores en el código.
* Uso de bibliotecas matemáticas como NumPy, Pandas, SymPy, Matplotlib, Plotly, Seaborn.



3. Pensamiento Computacional

Indicadores:

* Capacidad para descomponer problemas complejos en subproblemas manejables.
* Habilidad para diseñar algoritmos eficientes para resolver problemas específicos.
* Comprensión de la lógica detrás de las estructuras de control y su aplicación en la programación.
* Capacidad para abstraer conceptos matemáticos en soluciones programables.
* Habilidad para identificar y aplicar patrones en la resolución de problemas.



4. Aplicación de Matemáticas en Problemas Reales

Indicadores:

* Capacidad para modelar situaciones reales utilizando funciones matemáticas.
* Habilidad para implementar soluciones matemáticas en código Python para resolver problemas prácticos.
* Comprensión de cómo interpretar los resultados obtenidos mediante simulaciones o cálculos computacionales.
* Uso de Python para la visualización de datos y resultados matemáticos.
* Capacidad para integrar conocimientos matemáticos y de programación en proyectos interdisciplinarios.

## Cuestionario para realizar labores de memorización, observación y síntesis de la información (Además de diagnóstico)  

1. ¿Qué?
2. ¿Cómo?
3. ¿Dónde?
4. ¿Cuándo?
5. ¿Por qué? 

# Definiciones y Resultados científicos a comprender y memorizar 

[Primera grabación en Vocaroo]()

## Establecimiento de prerrequisitos conceptuales para alcanzar el objetivo de aprendizaje trazado.   

(Elabore una lista de preguntas al rededor de los conceptos que consideres necesarios para empezar la clase).  

1. 
2. 
3. 

# Para el diseño de la clase tenga en cuenta algunos problemas de aprendizaje

## Secuencia conceptual introductoria a la clase

## El componente histórico y epistemológico

# Vocabulary   

1. 
2.  
3.  
4.  
5.  

In [4]:
# Traducción del inglés al español con python

def traducir(texto):
    from googletrans import Translator

    translator = Translator() # instancia la clase 

    # Traducir al español
    translation = translator.translate(texto, src='en', dest='es')
    return translation.text

In [5]:
traducir('Hi')

'Hola'

In [ ]:
# mapa conceptual para cada clase  
from graphviz import Digraph

# Crear un nuevo grafo dirigido
dot = Digraph()

# Añadir nodos y relaciones al grafo
main_concept = ' '
dot.node('A', main_concept)
dot.node('B', '')
dot.node('C', '')
dot.node('D', '')

dot.edges(['AB'])


# Guardar el grafo como un archivo de imagen
dot.render(filename=main_concept, format='png', cleanup=True, \
           directory='C:/Users/marco/Documentos/docencia/')

dot.view()


'/home/marco/Documentos/docencia/calculo/images/0_reales/ .pdf'

# Cómo estimular el aprendizaje por experimentación para la adquisición de conceptos alrededor de los números enteros 

## Hipótesis de los estudiantes Luis Fernando y Cristián  

¿Será que los números naturales terminados en 1 siguen el patrón Primo-Primo- No primo.?  



In [1]:
def es_primo(n):
    if n < 2:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

In [5]:
import numpy as np 

numeros_que_terminan_en_cifra_1 = np.arange(11, 10_000, 10)
numeros_que_terminan_en_cifra_1 

array([  11,   21,   31,   41,   51,   61,   71,   81,   91,  101,  111,
        121,  131,  141,  151,  161,  171,  181,  191,  201,  211,  221,
        231,  241,  251,  261,  271,  281,  291,  301,  311,  321,  331,
        341,  351,  361,  371,  381,  391,  401,  411,  421,  431,  441,
        451,  461,  471,  481,  491,  501,  511,  521,  531,  541,  551,
        561,  571,  581,  591,  601,  611,  621,  631,  641,  651,  661,
        671,  681,  691,  701,  711,  721,  731,  741,  751,  761,  771,
        781,  791,  801,  811,  821,  831,  841,  851,  861,  871,  881,
        891,  901,  911,  921,  931,  941,  951,  961,  971,  981,  991,
       1001, 1011, 1021, 1031, 1041, 1051, 1061, 1071, 1081, 1091, 1101,
       1111, 1121, 1131, 1141, 1151, 1161, 1171, 1181, 1191, 1201, 1211,
       1221, 1231, 1241, 1251, 1261, 1271, 1281, 1291, 1301, 1311, 1321,
       1331, 1341, 1351, 1361, 1371, 1381, 1391, 1401, 1411, 1421, 1431,
       1441, 1451, 1461, 1471, 1481, 1491, 1501, 15

In [6]:
[es_primo(n) for n in numeros_que_terminan_en_cifra_1] 

[True,
 False,
 True,
 True,
 False,
 True,
 True,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 True,
 False,
 False,
 True,
 True,
 False,
 True,
 False,
 False,
 True,
 True,
 False,
 True,
 True,
 False,
 False,
 True,
 False,
 True,
 False,
 False,
 False,
 False,
 False,
 False,
 True,
 False,
 True,
 True,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 True,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 False,
 True,
 True,
 False,
 True,
 False,
 False,
 True,
 True,
 False,
 False,
 False,
 False,
 True,
 True,
 False,
 False,
 False,
 False,
 True,
 True,
 False,
 False,
 False,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 True,
 False,
 False,
 True,
 True,
 False,
 True,
 True,
 False,
 False,
 True,
 False,
 False,
 False,
 False,
 False,
 True,
 False,
 True,
 True,
 False,
 True,
 False,
 False,
 True,
 False,
 False,
 False,
 False,
 False,
 True,
 True,
 False,
 Tru

## Conclusión 
Se descarta la hipótesis de que los números naturales terminados en 1 siguen el pantron P-P-NP

# Ejemplo ilustrativo

# 10 ejercicios de entrenamiento o adquisición de habilidades obtenido a través de ChatGPT para una formación personalizada y centrada en la habilidad que se quiere lograr con la enseñanza del ejemplo anterior

Esta es una colección de 10 ejercicios de conjeturas sobre números enteros que pueden probar y validar usando la función `es_primo(n)`. Cada conjetura les permitirá explorar patrones y características de los números primos y otros tipos de números enteros.



1. **Conjetura de Números Gemelos**  
   Investigar si existen números primos gemelos para los primeros \( N \) números naturales, donde un par de números primos gemelos son primos consecutivos que tienen una diferencia de 2 (como 11 y 13).  
   **Ejemplo**: Validar que \( 17 \) y \( 19 \) son primos gemelos.

2. **Conjetura de Números Primoriales**  
   Explorar si todos los primos de la forma \( p_n! + 1 \) (factorial de un número primo más 1) son números primos.  
   **Ejemplo**: Probar con $p = 5$: \( 5! + 1 = 121 \).



3. **Conjetura de los Números de Mersenne**  
   Comprobar si los números de la forma \( 2^p - 1 \) (donde \( p \) es un número primo) son siempre primos. Estos números se conocen como números de Mersenne.  
   **Ejemplo**: Verificar si \( 2^3 - 1 = 7 \) es primo.



4. **Conjetura de Primos de la Forma \( n^2 + n + 41 \)**  
   Evaluar si la expresión \( n^2 + n + 41 \) produce números primos para \( n \) entre 0 y 39.  
   **Ejemplo**: Comprobar si \( 0^2 + 0 + 41 = 41 \) y \( 1^2 + 1 + 41 = 43 \) son primos.



5. **Conjetura de Primos de Fermat**  
   Verificar si los números de la forma \( 2^{2^n} + 1 \) son primos para $n = 0, 1, 2, 3, \ldots$.  
   **Ejemplo**: Comprobar si \( 2^{2^2} + 1 = 17 \) es primo.



6. **Conjetura de Números Semi-Primos**  
   Probar si un número compuesto se puede expresar siempre como el producto de dos números primos.  
   **Ejemplo**: Comprobar si $15 = 3 \times 5$.



7. **Conjetura de Primos de Sophie Germain**  
   Investigar si, dado un número primo \( p \), el número \( 2p + 1 \) también es primo. Estos números \( p \) se llaman primos de Sophie Germain.  
   **Ejemplo**: Comprobar si, siendo \( p = 5 \), \( 2 \times 5 + 1 = 11 \) es primo.



8. **Conjetura sobre Suma de Dos Primos**  
   Explorar si todo número par mayor que 2 puede expresarse como la suma de dos números primos (Conjetura de Goldbach).  
   **Ejemplo**: Comprobar si $10 = 3 + 7$ y si $18 = 5 + 13$.



9. **Conjetura de Números Triangulares y Primos**  
   Investigar si el número triangular \( T_n = \frac{n(n+1)}{2} \) es primo para algunos valores de $n$.  
   **Ejemplo**: Comprobar si \( T_2 = 3 \) y \( T_3 = 6 \) son primos.



10. **Conjetura de Números Cuadrados y Primos**  
    Comprobar si existe algún número primo que también sea un número cuadrado perfecto.  
    **Ejemplo**: Probar si \( 9 = 3^2 \) y es primo.

Cada uno de estos ejercicios fomenta la investigación y el razonamiento matemático en torno a las propiedades de los números enteros, permitiendo a tus estudiantes poner a prueba sus habilidades en la programación y en el análisis de patrones numéricos.

## Tareas para realizar antes de la clase  

1. Ver los videos respectivos []()
2. Memorizar las definiciones []()
3. Sintetizar o resumir la información en mapas conceptuales. []()
4. Tareas para incentivar el uso de la calculadora Casio 991LA cw Para ingenieros, licenciados y administradores.  
5. Leer los ejemplos resueltos y replicar la manera de hacerlos

In [1]:
# Tarea 1 de lectura de ejemplos y completación de procedimientos
# asignación de tarea para realizar antes de clase
# Consiste en leer y completar un ejemplo de texto guía
# Esta tarea es para realizar solo de manera manuscrita en su cuaderno de clase. 
# La otra opción de presentación es elaborada en cuadernos Jupyter (En este caso solo debe realizar la cuarta parte de los ejercicios asignados)
# 
def asignador_de_tareas(nombre_tarea, ubicacion_tarea):
    import pandas as pd
    import numpy as np
    import requests
    np.random.seed(1213)  # establecer semilla aleatoria para hacer los resultados reproducibles

    def asignar_tarea(nombre_tarea, grupo_estudiantes, nombre_texto_guia, pag_seccion, numero_ejemplos_seccion, numero_ejemplos_asignados):
        students_path = "/home/marco/Documentos/docencia/groups_list/" + grupo_estudiantes + ".csv"
        examples = list(range(1, numero_ejemplos_de_la_seccion))  
        students_data = pd.read_csv(students_path)
        # Crear una lista para almacenar las asignaciones
        assignments = []

        # Asignar ejercicios aleatorios a cada estudiante
        for index, student in students_data.iterrows():
            # Seleccionar 5 ejercicios aleatorios sin repetición
            random_exercises = np.sort(np.random.choice(examples, numero_de_ejemplos_asignados, \
                                                replace = False)+ 1)
            assignments.append([students_data["Nombre"][index], *random_exercises])  # Agregar asignación a la lista

    # Crear un DataFrame con las asignaciones
        columns = ["Student"] + [ "Example" + str(i+1) for i in range(numero_de_ejemplos_asignados)]
        assignment_df = pd.DataFrame(assignments, columns=columns)

    # Guardar el DataFrame en un archivo CSV
        assignment_csv = nombre_tarea + "_" + nombre_texto_guia + "_"+ pagina_y_seccion + ".csv"
        assignment_df.to_csv(assignment_csv, index=False)

        print("""Tarea 1 de lectura de ejemplos y completación de procedimientos: 
     Si faltan los enunciados de los problemas, no se valora la tarea
    Si faltan de la conclusión verbales, no se valora la tarea.
    Esta tarea se valora solo si la presenta de manera manuscrita en su cuaderno de clase.  
      """, assignment_csv)
        return assignment_df

# Componente de manejo de calculadora Casio asociado al objetivo de aprendizaje de esta clase  

## Pasos para lograr el objetivo de aprendizaje con la experimentación en la calculadora Casio 991LA cw  

In [2]:
asignador_de_tareas(nombre_tarea, ubicacion_tarea)



NameError: name 'nombre_tarea' is not defined

In [ ]:
asignador_de_tareas(nombre_tarea, ubicacion_tarea):





Tarea 3 Solución problemas describiendo y justificando verbalmente: 
      Si faltan los enunciados de los problemas, no se valora la tarea
      Si faltan de la conclusión verbales, no se valora la tarea 
      Esta tarea se valora solo si la presenta de manera manuscrita en su cuaderno de clase  
       tarea_Purcell_Página_202_Sección_3.8.csv


Student  Exer1  Exer2  Exer3  Exer4  Exer5  \
0   AMARIS FERNANDEZ SEBASTIAN CAMILO      4      7      9     17     21   
1       ARENAS HERAZO GABRIEL ENRIQUE      2      5     11     17     26   
2        AYALA MENDOZA MARISOL YACIRA     10     18     21     37     39   
3          BERTEL RODELO MICHELL SMIT      4      8     20     26     28   
4            BULA VERGARA MARIA PAULA     12     13     20     23     25   
5             CANTILLO ENSUCHO DANIEL      2      4      5      7     15   
6            CERVANTES DURAN ANGELICA      2      4      5     14     17   
7             GUERRA NORIEGA YONATHAN      3     20     24     26     28   
8     HINCAPIE CASTRO JOHAN SEBASTIAN     10     17     27     28     29   
9              HUMANEZ SANCHEZ CAMILO      3     16     19     24     33   
10       IBAÑEZ MORENO FRANCISCO LUIS      3      8     13     16     21   
11          MASS MIRANDA RIVALDO JOSE      2      3     12     13     23   
12     MIRANDA MARTINEZ CARLOS ANDRES     10     12     21     24     25   
13              MONTERROSA SOTO ABRIL     10     14     16     18     28   
14           NORIEGA JIMENEZ GABRIELA      4     13     14     16     19   
15             PEREZ REYES IVAN DARIO     17     19     23     26     30   
16               PEREZ THERAN ERLINDA      3      6     19     21     25   
17              PINEDA ROMERO YOSMERI      2     30     31     32     35   
18         QUINONEZ CERVANTES MARIANA      5      6      8     13     15   
19                 RAMOS PEREZ YANINE     11     20     21     28     29   
20              RIVAS HOYOS ESTEFANIA      7     10     12     14     17   
21              ROSILLO TEJADA VANESA     12     22     23     27     28   
22          SANCHEZ PADILLA SILVESTRE      4      7     15     22     32   
23        SILVA ROYERO MAILY CRISTINA      4      5     12     13     20   
24          USUGA GEORGE JHON ESTIVEN      5      7      8     10     22   

    Exer6  Exer7  Exer8  Exer9  Exer10  
0      30     39     44     51      54  
1      28     32     40     46      50  
2      45     46     47     51      52  
3      33     40     41     42      47  
4      27     29     30     34      48  
5      25     27     33     46      51  
6      19     22     24     34      52  
7      35     39     48     49      53  
8      32     36     44     48      54  
9      41     43     44     47      49  
10     27     28     31     42      43  
11     24     31     35     46      52  
12     26     27     29     39      50  
13     30     35     36     42      50  
14     22     36     37     46      51  
15     33     34     40     46      53  
16     30     32     39     42      45  
17     36     37     42     47      50  
18     32     33     42     46      49  
19     37     39     44     46      52  
20     25     28     30     37      48  
21     30     34     36     40      44  
22     34     38     43     48      53  
23     25     30     35     42      54  
24     37     39     42     44      51

# Horario de clases del profesor Marco

* Lunes 8-12: Cálculo Vectorial
* Martes 8-12 M: Fundamentos de Aritmética. 
* Miércoles 8 a 12 M: Fundamentos de la Lógica
* Jueves 2 a 6 PM: Desarrollo del pensamiento matemático en la infancia. 
* Viernes 8 - 12 M: Desarrollo del pensamiento matemático en la infancia. 
* Sábados 8-12 Asesorías y espacio para retroalimentación y apoyo al trabajo independiente. 

# Horarios de monitoría y asesoría  

Coordinador de los cursos de la Facultad de Educación para regiones:    

Andrés Vélez: regioneducacion.fedu@udea.edu.co  
Coordinador Regiones  
Facultad de Educación  
Universidad de Antioquia  

## Monitor Manuel San Juan Serrano

Contactar escribiendo al correo: manuel.serrano1@udea.edu.co

Martes y jueves de 8 - 12  
Miércoles 10 - 12 y de 2 a 4 PM  
Sábado: 8 - 12

## Monitoría virtual   

Telefono del monitos en whatsApp: 3043936088  

Juan David Cano Castrillón  
Estudiante de Pregrado.  
juan.canoc1@udea.edu.co  
3043936088  


## Referentes 

* [Grajales Vanegas, L. M., Restrepo Estrada, C. E., Restrepo Ochoa, S. I., & Ruíz De Villalba, F. (2015). Matemáticas I para las ciencias económicas.](https://github.com/marco-canas/fundamentos_de_la_aritmetica/blob/main/referentes/2_grajales_logica/grajales_lina_matemticas.pdf)
  
* Duval, R. (2004). Semiosis y pensamiento humano: registros semióticos y aprendizajes intelectuales. Universidad del Valle.

* [Aylwin, C. U. (2011). Lógica, conjuntos y números. Universidad de los Andes, Consejo de Publicaciones, Colección: Ciencias Básicas, Serie: Matemáticas.](https://github.com/marco-canas/fundamentos_logica/blob/main/referentes/2_uzcategui/uscategui.pdf)
  
* [Chollet, F. (2021). Deep learning with Python. Simon and Schuster.](https://github.com/marco-canas/didactica_ciencia_datos/blob/main/2_referentes/chollet/Fran%C3%A7ois%20Chollet%20-%20Deep%20Learning%20with%20Python-Manning%20(2018).pdf)  
  
* [Watson, S., Stewart, J., & Redlin, L. (2009). Precálculo. Matemáticas para el cálculo.](https://github.com/marco-canas/calculo/blob/main/referents/1_precalculo_stewart/stewart_precalculo.pdf)  

* [Purcell, E. J., Varberg, D., & Rigdon, S. E. (2007). Cálculo diferencial e integral. Pearson Educación.](https://github.com/marco-canas/calculo/blob/main/referents/purcell/purcell_calculo.pdf)

  

* [stewart cálculo](https://udeaeduco-my.sharepoint.com/:b:/g/personal/marco_canas_udea_edu_co/EZgXZjAp8QxPqOAim2hs6LcBNPLGjSHf-xwYnUVYkwa04w?e=RZdTCy)  

* [larson](https://udeaeduco-my.sharepoint.com/:b:/g/personal/marco_canas_udea_edu_co/ES71ChFeO9ZDhW3TwC5Ijk8BjxUK3Pdqz_fjHxTTFAfIAg?e=VDEjfu)



* [Recomendación de la UNESCO sobre ciencia abierta](https://unesdoc.unesco.org/ark:/48223/pf0000379949_spa)

* [chatGPT](https://openai.com/blog/chatgpt)  

* [Géron, A. (2017). Hands-on machine learning with scikit-learn and tensorflow: Concepts. Tools, and Techniques to build intelligent systems.](https://github.com/marco-canas/didactica_ciencia_datos/blob/main/2_referentes/geron/Hands-On%20Machine%20Learning-3nd_Edition(2022).pdf)  

* [McKinney, W. (2012). Python for data analysis: Data wrangling with Pandas, NumPy, and IPython. " O'Reilly Media, Inc.".](https://github.com/marco-canas/didactica_ciencia_datos/blob/main/2_referentes/mckinney/Wes_McKinney.pdf)  

# Como estudiante, encuentro que...   

F: Mis Fortalezas son:     
O: Mis Oportunidades son:    
D: Mis Debilidades son:    
A: Lo que Amenazas mi aprendizaje es:  

### [Evaluamos al profesor Marco Cañas Aquí](https://forms.office.com/Pages/ResponsePage.aspx?id=IefhmYRxjkmK_7KtTlPBwkanXIs1i1FEujpsZgO6dXpUREJPV1kxUk1JV1ozTFJIQVNIQjY5WEY3US4u)

### Continue su aprendizaje en la siguiente clase a través del siguiente [vínculo]()

### **Actividad de Clase: Expresar Ideas, Dudas y Emociones de Manera Respetuosa y Asertiva**



**Objetivo:**  
Desarrollar habilidades de comunicación efectiva en los estudiantes para que puedan expresar sus ideas, dudas y emociones de forma respetuosa y asertiva.



**Duración:**  
15-20 minutos

---



### **1. Introducción (5 minutos)**  
- **Tema:** Introducir a los estudiantes al concepto de la comunicación asertiva, explicando cómo expresar pensamientos y emociones de manera clara y respetuosa sin agredir o ser pasivo.
- **Dinámica:** Mostrar dos ejemplos contrastantes: uno de comunicación pasiva/agresiva y otro de comunicación asertiva. Preguntar a los estudiantes cuál creen que es más efectivo y por qué.



**Materiales:**  
- Pizarra o diapositivas con ejemplos de frases pasivas, agresivas y asertivas.
  
**Ejemplo:**  
- **Pasivo:** “Bueno… no sé si es una buena idea, pero… tal vez podrías intentarlo.”
- **Agresivo:** “¡Tu idea es terrible, no va a funcionar!”
- **Asertivo:** “Creo que podríamos mejorar esa idea si consideramos otro enfoque.”

---



### **2. Ejercicio en Parejas (10 minutos)**  
- **Instrucciones:** Dividir a los estudiantes en parejas. Darles diferentes situaciones cotidianas en las que deben comunicarse (pedir ayuda, expresar desacuerdo, compartir una idea, etc.). Cada pareja debe crear una breve conversación donde practiquen la comunicación asertiva.
  
**Situaciones propuestas:**
  - Expresar desacuerdo con un compañero sobre una tarea de grupo.
  - Pedirle ayuda al profesor sobre un concepto difícil.
  - Expresar frustración por un malentendido en clase.

- **Reglas:** Cada estudiante debe asegurarse de usar lenguaje respetuoso y asertivo para expresar su posición y emociones. Al final, cada pareja debe compartir con la clase un ejemplo de lo que discutieron.

---



### **3. Retroalimentación y Discusión (5 minutos)**  
- **Discusión:** Invitar a algunos estudiantes a compartir sus experiencias de la actividad. 
- **Preguntas:** ¿Cómo se sintieron al intentar ser asertivos? ¿Les resultó difícil? ¿Cómo fue recibido el mensaje?
  
**Consejo:** Resaltar la importancia de practicar la asertividad como una herramienta para mejorar la comunicación interpersonal y reducir conflictos.

---



### **Cierre**  
- Resumir las características de la comunicación asertiva: uso de "yo" en lugar de "tú" para evitar culpabilizar, tono de voz calmado, y respeto hacia el interlocutor.
- **Tarea opcional:** Pedirles que practiquen la comunicación asertiva en sus conversaciones durante la semana y escriban una reflexión breve sobre cómo les fue.



**Material adicional:**  
- Video corto o recurso digital sobre la importancia de la comunicación asertiva.

## Agradecimientos  

Doy gracias a Dios por la vida de mi Hijo Joseph Cañas Osorio y la madurez que ha alcanzado. Este hijo me enorgullece y me hace falta abrazarlo cada día. 

Y a mi esposa Yasmira Emperatriz Barboza Mogollón por su apoyo, orientación y acompañamiento. 